core/cautils/git_native_disabled.go

```diff
@@ -3,20 +3,85 @@
 package cautils
 
 import (
-	"errors"
+	"fmt"
+	"time"
 
 	"github.com/kubescape/go-git-url/apis"
+	"github.com/kubescape/go-logger"
+	"github.com/matthyx/go-gitlog"
 )
 
-var ErrFatalNotSupportedByBuild = errors.New(`git scan not supported by this build. Build with tag "gitenabled" to enable the git scan feature`)
-
 type gitRepository struct {
+	gitRepo          gitlog.GitLog
+	fileToLastCommit map[string]*gitlog.Commit
 }
 
 func newGitRepository(root string) (*gitRepository, error) {
-	return &gitRepository{}, ErrWarnNotSupportedByBuild
+	gitRepo := gitlog.New(&gitlog.Config{Path: root})
+
+	return &gitRepository{
+		gitRepo: gitRepo,
+	}, nil
 }
 
 func (g *gitRepository) GetFileLastCommit(filePath string) (*apis.Commit, error) {
-	return nil, ErrFatalNotSupportedByBuild
+	if len(g.fileToLastCommit) == 0 {
+		g.buildCommitMap()
+	}
+
+	if relevantCommit, exists := g.fileToLastCommit[filePath]; exists {
+		return g.getCommit(relevantCommit), nil
+	}
+
+	return nil, fmt.Errorf("failed to get commit information for file: %s", filePath)
+}
+
+func (g *gitRepository) buildCommitMap() {
```

Comment:
There is no need for `//go:build !gitenabled` anymore. So you do not need to duplicate the code in both files.
Also, we can update the installation and remove the git2lib from the installation (and use `git build` instead)

Reply:
yes, but maybe for the moment we can build both and make sure it works for all users